plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,284 @@
+        return new SelectionAdapter() {
+
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                FileDialog dialog = createFileDialog(model.getLocalSourceFileName());
+                String file = dialog.open();
+                setLocalSourceFileName(file);
+            }
+
+            private FileDialog createFileDialog(String selectedFile) {
+                FileDialog dialog = new FileDialog(getShell(), SWT.OPEN);
+                dialog.setText("Select an OpenShift resource");
+                if(isFile(selectedFile)) {
+                    File file = new File(selectedFile);
+                    dialog.setFilterPath(file.getParentFile().getAbsolutePath());
+                }
+                return dialog;
+            }
+        };
+    }
+    
+    private SelectionListener onBrowseWorkspaceClicked() {
+        return new SelectionAdapter() {
+
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                ElementTreeSelectionDialog dialog = UIUtils.createFileDialog(model.getLocalSourceFileName(),
+                                                                             "Select an OpenShift resource",
+                                                                             "Select an OpenShift resource (*.json)",
+                                                                             "json",
+                                                                             null);
+                if (dialog.open() == IDialogConstants.OK_ID && dialog.getFirstResult() instanceof IFile) {
+                    String path = ((IFile)dialog.getFirstResult()).getFullPath().toString();
+                    String file = VariablesHelper.addWorkspacePrefix(path);
+                    setLocalSourceFileName(file);
+                }
+            }
+
+
+        };
+    }
+    
+    private void setLocalSourceFileName(String file) {
+        if (file == null || !isFile(file)) {
+            return;
+        }
+        try {
+            model.setLocalSourceFileName(file);
+        } catch (ClassCastException | OpenShiftException ex) {
+            IStatus status = ValidationStatus.error(ex.getMessage(), ex);
+            OpenShiftUIActivator.getDefault().getLogger().logStatus(status);
+            ErrorDialog.openError(getShell(), "Openshift resource error",
+                    NLS.bind("The file \"{0}\" is not an OpenShift resource.", 
+                            file),
+                    status);
+        }
+    }
+   
+    private void createRemoteSourceControls(DataBindingContext dbc, Group sourceGroup) {
+        Composite composite = new Composite(sourceGroup, SWT.NONE);
+        GridLayoutFactory.fillDefaults().numColumns(2).applyTo(composite);
+        GridDataFactory.fillDefaults().span(3, 1).grab(true,  false).applyTo(composite);
+        Button remoteButton = new Button(composite, SWT.RADIO);
```

Comment:
New Java Project, as an example, places radio and relevant input control in one line. Can we follow the same pattern? Also, why not remoteButton.setText("Create from url") but a separate label?